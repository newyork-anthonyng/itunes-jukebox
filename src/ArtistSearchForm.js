import React from "react";
import PropTypes from "prop-types";

const ArtistSearchForm = ({ onSubmit, onChange, value }) => (
  <form onSubmit={onSubmit}>
    <label>
      Search albums for artist
      <input
        type="text"
        placeholder="Enter artist name"
        value={value}
        onChange={onChange}
      />
    </label>

    <button type="submit">Search</button>
  </form>
);

ArtistSearchForm.propTypes = {
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  value: PropTypes.string
};

export default ArtistSearchForm;
