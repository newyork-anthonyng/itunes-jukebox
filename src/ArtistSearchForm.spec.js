import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import ArtistSearchForm from "./ArtistSearchForm";

const defaultProps = {
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  value: "Michael Jackson"
};

it("should render correctly", () => {
  const wrapper = shallow(<ArtistSearchForm {...defaultProps} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should call correct function when submitting form", () => {
  const cb = jest.fn();
  const wrapper = mount(<ArtistSearchForm {...defaultProps} onSubmit={cb} />);

  const button = wrapper.find("button");
  // https://github.com/airbnb/enzyme/issues/308#issuecomment-291604063
  button.simulate("submit");

  expect(cb).toHaveBeenCalledTimes(1);
});

it("should call correct function when changing input", () => {
  const cb = jest.fn();
  const wrapper = shallow(<ArtistSearchForm {...defaultProps} onChange={cb} />);

  const input = wrapper.find("input");
  input.simulate("change", { target: { value: "Michael " } });

  expect(cb).toHaveBeenCalledTimes(1);
  expect(cb.mock.calls[0]).toMatchSnapshot();
});
