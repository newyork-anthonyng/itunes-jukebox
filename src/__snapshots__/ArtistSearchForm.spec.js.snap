// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call correct function when changing input 1`] = `
Array [
  Object {
    "target": Object {
      "value": "Michael ",
    },
  },
]
`;

exports[`should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Search albums for artist
    <input
      onChange={[Function]}
      placeholder="Enter artist name"
      type="text"
      value="Michael Jackson"
    />
  </label>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
